verilog/tools/kythe/indexing_facts_tree_extractor.cc

```diff
@@ -87,45 +89,52 @@ IndexingFactNode ExtractOneFile(absl::string_view content,
   const auto& syntax_tree = text_structure.SyntaxTree();
 
   return BuildIndexingFactsTree(syntax_tree, analyzer.Data().Contents(),
-                                filename, file_list_facts_tree,
-                                extracted_files);
+                                filename, file_list_facts_tree, extracted_files,
+                                include_dir_paths);
 }
 
-// Returns the path of the file relative to the file list.
-std::string GetPathRelativeToFileList(absl::string_view file_list_dir,
-                                      absl::string_view filename) {
-  return verible::file::JoinPath(file_list_dir, filename);
+// Tries to read the files in all the given directories.
+// Returns the first file it find in case of many files with the same name.
+bool TryToReadFile(std::string& file_path, std::string& content,
+                   absl::string_view filename,
+                   const std::set<std::string>& directories) {
+  for (auto& dir_path : directories) {
+    file_path = verible::file::JoinPath(dir_path, filename);
+    if (verible::file::GetContents(file_path, &content).ok()) {
+      return true;
+    }
+  }
+  return false;
 }
 
 }  // namespace
 
 IndexingFactNode ExtractFiles(const std::vector<std::string>& ordered_file_list,
-                              int& exit_status,
-                              absl::string_view file_list_dir) {
+                              int& exit_status, absl::string_view file_list_dir,
+                              const std::set<std::string>& include_dir_paths) {
   // Create a node to hold the dirname of the ordered file list and group all
   // the files and acts as a ordered file list of these files.
   IndexingFactNode file_list_facts_tree(IndexingNodeData(
       {Anchor(file_list_dir, 0, 0)}, IndexingFactType::kFileList));
-  std::set<std::string> extracted_files;
+  std::map<std::string, std::string> extracted_files;
 
   for (const auto& filename : ordered_file_list) {
-    std::string file_path = GetPathRelativeToFileList(file_list_dir, filename);
-
     // Check if this included file was extracted before.
-    if (extracted_files.find(file_path) != extracted_files.end()) {
+    if (extracted_files.find(filename) != extracted_files.end()) {
       continue;
     }
-    extracted_files.insert(file_path);
 
+    std::string file_path;
     std::string content;
-    if (!verible::file::GetContents(file_path, &content).ok()) {
-      LOG(ERROR) << "Error while reading file: " << file_path;
-      exit_status = 1;
+    if (!TryToReadFile(file_path, content, filename, include_dir_paths)) {
+      LOG(ERROR) << "Error while reading file: " << filename;
+      // exit_status = 1;
```

Comment:
In case we failed to read one file, the extractor doesn't work.
I think we should remove this `exit_status`.